### PR TITLE
PI-2383 - add arrival and departure date to contact record

### PR DESCRIPTION
**Pull request by achimber-moj**

### CODE REVIEW

File: projects/approved-premises-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/ReferralService.kt [hunk 1]
```diff
@@ -70,7 +70,15 @@ class ReferralService(
...
-                        notes = "To view details of the Approved Premises booking, click here: ${details.applicationUrl}",
+                        notes = listOfNotNull(
+                            "Current: ${DeliusDateFormatter.format(details.arrivalOn)} to ${
+                                DeliusDateFormatter.format(
+                                    details.departureOn
+                                )
+                            }",
+                            "To view details of the Approved Premises booking, click here: ${details.applicationUrl}"
+                        )
+                            .joinToString(System.lineSeparator() + System.lineSeparator()),
```
Comment (marcus-bcl) [inline projects/approved-premises-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/ReferralService.kt:81]: Neither of these strings can be null, so might be clearer as a multi-line string?
```suggestion
                        notes = """
                            Current: ${DeliusDateFormatter.format(details.arrivalOn)} to ${DeliusDateFormatter.format(details.departureOn)}
                            
                            To view details of the Approved Premises booking, click here: ${details.applicationUrl}
                            """.trimIndent(),
```

File: projects/approved-premises-and-delius/src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/MessagingIntegrationInactiveTest.kt [hunk 2]
```diff
@@ -150,7 +150,13 @@ internal class MessagingIntegrationInactiveTest {
         assertThat(contact.description, equalTo("Approved Premises Booking for Hope House"))
         assertThat(
             contact.notes,
-            equalTo("To view details of the Approved Premises booking, click here: https://approved-premises-dev.hmpps.service.justice.gov.uk/applications/484b8b5e-6c3b-4400-b200-425bbe410713")
+            equalTo(
+                """
+                Current: 30/01/2023 to 30/04/2023
```
Comment (marcus-bcl) [inline projects/approved-premises-and-delius/src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/MessagingIntegrationInactiveTest.kt:155]: Has this format come from somewhere, or is it a best guess?

Just wondering if we should make it clearer/more consistent with other generated notes fields e.g. something like:
```suggestion
                Expected arrival: 30/01/2023
                Expected departure: 30/04/2023
```

Comment (achimber-moj) [reply projects/approved-premises-and-delius/src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/MessagingIntegrationInactiveTest.kt:155]: I took it from Booking changed note format.

Comment (achimber-moj) [reply projects/approved-premises-and-delius/src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/MessagingIntegrationInactiveTest.kt:155]: I will make the change to separate the fields.

**Review by marcus-bcl — APPROVED** [review]

**Pull request merged**